src/backend/commands/extension.c

```diff
@@ -1242,6 +1244,16 @@ identify_update_path(ExtensionControlFile *control,
 	/* Find shortest path */
 	result = find_update_path(evi_list, evi_start, evi_target, false, false);
 
+	/* Before we report an ERROR, try to download a remote extension */
+	if (result == NIL && download_extension_file_hook && !attempted_download)
+	{
+		attempted_download = true;
+		download_extension_file_hook(control->name, false);
+
+		/* Try again to find the shortest path */
+		goto reidentify;
```

Comment:
I recall that in one of the previous versions you did just
```
result = find_update_path(evi_list, evi_start, evi_target, false, false);
```
here. Why did you decide to don't go with this approach? Were there any problems with it?

Reply:
Ah, probably some file that we previously read may change after downloading?

Reply:
Postgres gets the version "tree" where I added the label. All the PG PRs have the same change. You had just approved an outdated diff on the first one. The change was a little larger than I had liked unfortunately.